homeassistant/components/renault/sensor.py

```diff
@@ -45,224 +70,233 @@ async def async_setup_entry(
-
-
-class RenaultFuelQuantitySensor(RenaultCockpitDataEntity, SensorEntity):
-    """Fuel quantity sensor."""
-
-    _attr_icon = "mdi:fuel"
-    _attr_native_unit_of_measurement = VOLUME_LITERS
-
-    @property
-    def native_value(self) -> int | None:
-        """Return the state of this entity."""
-        if not self.data or self.data.fuelQuantity is None:
-            return None
-        return round(self.data.fuelQuantity)
+            return cast(float, raw_value) / 1000
+        return cast(float, raw_value)
 
 
-class RenaultMileageSensor(RenaultCockpitDataEntity, SensorEntity):
-    """Mileage sensor."""
-
-    _attr_icon = "mdi:sign-direction"
-    _attr_native_unit_of_measurement = LENGTH_KILOMETERS
-
-    @property
-    def native_value(self) -> int | None:
-        """Return the state of this entity."""
-        if not self.data or self.data.totalMileage is None:
-            return None
-        return round(self.data.totalMileage)
-
-
-class RenaultOutsideTemperatureSensor(RenaultHVACDataEntity, SensorEntity):
-    """HVAC Outside Temperature sensor."""
-
-    _attr_device_class = DEVICE_CLASS_TEMPERATURE
-    _attr_native_unit_of_measurement = TEMP_CELSIUS
-
-    @property
-    def native_value(self) -> float | None:
-        """Return the state of this entity."""
-        return self.data.externalTemperature if self.data else None
-
-
-class RenaultPlugStateSensor(RenaultBatteryDataEntity, SensorEntity):
+class RenaultBatteryPlugStateSensor(RenaultBatterySensor):
     """Plug State sensor."""
 
-    _attr_device_class = DEVICE_CLASS_PLUG_STATE
-
     @property
     def native_value(self) -> str | None:
         """Return the state of this entity."""
-        plug_status = self.data.get_plug_status() if self.data else None
-        return plug_status.name.lower() if plug_status is not None else None
-
-    @property
-    def icon(self) -> str:
-        """Icon handling."""
-        return "mdi:power-plug" if self.is_plugged_in else "mdi:power-plug-off"
+        plug_status = (
+            self.coordinator.data.get_plug_status() if self.coordinator.data else None
+        )
+        return plug_status.name.lower() if plug_status else None
+
+
+SENSOR_TYPES: tuple[RenaultSensorEntityDescription, ...] = (
+    RenaultSensorEntityDescription(
+        key="battery_level",
+        coordinator="battery",
+        data_key="batteryLevel",
+        device_class=DEVICE_CLASS_BATTERY,
+        entity_class=RenaultBatterySensor,
+        name="Battery Level",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    RenaultSensorEntityDescription(
+        key="charge_state",
+        coordinator="battery",
+        data_key="chargingStatus",
+        device_class=DEVICE_CLASS_CHARGE_STATE,
+        entity_class=RenaultBatteryChargeStateSensor,
+        icon_lambda=lambda x: "mdi:flash"
+        if x.data == ChargeState.CHARGE_IN_PROGRESS.value
+        else "mdi:flash-off",
```

Comment:
This is quite difficult to read. Try adding brackets. That way black might format it a bit better.
```py
icon_lambda=lambda x: (...),